### ISSUE TITLE
Including PLSQL 

### ISSUE THREAD

**Issue by MariusKochFHDW** [GumTreeDiff/gumtree#359]
Hey there,

i was just set on a new Project to find a difftool for PLSQL and i recently found gumtree. I already have the antlr4 grammar and parser for PLSQL and have a python script to generate an AST. I already seen the tree-sitter but can't really figure out how to do it. 

Thank you in advance :)

**Comment by jrfaller** [comment]
Hi! Can you find the parser you want here : https://github.com/bonede/tree-sitter-ng ? Cheers.

**Comment by MariusKochFHDW** [comment]
No, unfortunatly there isn't any supported parser for this programming language.

**Comment by jrfaller** [comment]
Is there any tree-sitter parser you know that might do the trick ?

**Comment by MariusKochFHDW** [comment]
I dont see any.

**Comment by jrfaller** [comment]
so maybe you try the antlr-4 integration but I have never really used it seriously

**Comment by jrfaller** [comment]
We have a helper here -> https://github.com/GumTreeDiff/gumtree/blob/main/gen.antlr4/src/main/java/com/github/gumtreediff/gen/antlr4/AbstractAntlr4TreeGenerator.java But I have no sample project to showcase how to use it 😢

**Comment by MariusKochFHDW** [comment]
Thank you for the response :)
I will have a look at it and tell you if it works. 

Hoping for the best 😄

**Comment by jrfaller** [comment]
Maybe look at how we do for antlr 3 projects like this -> https://github.com/GumTreeDiff/gumtree/tree/main/gen.antlr3-r

**Comment by MariusKochFHDW** [comment]
Hey there again!

I just tried a bunch of stuff and came to the conclusion that the best/easiest way for me to implement PLSQL into Gumtree is the following:
1. I'm using the antlr4 Lexer and Parser to generate a AST in the form of the antlr4
2. From there i would love to just send it to either the API or a Method that can handle the antlr4 AST and format it into the XML format gumtree can handle.

Is there some kind of Method or API call that does the trick?
Im pretty new to coding so i hope i didnt overlooked anything importent.
Thankss :)

**Comment by jrfaller** [comment]
Hi! Normally if you have a antlr4 grammar you should be able to skip the XML part and use directly the antlr generated objects to convert them to a GumTree AST it would be much more efficient?

**Comment by jrfaller** [comment]
Hi! Once you have converted your parser specific format to GumTree's agnostic format (a `Tree` and a `TreeContext`) every other things come for free (such as the diff :D). For `antlr3` grammars it is done via subclassing our helper class like here : https://github.com/GumTreeDiff/gumtree/blob/main/gen.antlr3-r/src/main/java/com/github/gumtreediff/gen/antlr3/r/RTreeGenerator.java . I have no such example for antlr4 grammars but as far as I can see you have 3 methods to override here (and also don't forget to add the annotation to register the parser, as done in the antlr3 example). For the webdiff command it's trickier because you have to include the antlr4 projects in the distribution here : https://github.com/GumTreeDiff/gumtree/blob/main/dist/build.gradle . Sorry this a lot of stuff to do 😢

**Comment by MariusKochFHDW** [comment]
Hey man, almost everything done!
I created a antlr3 folder and put an old grammar in there and registered everything.
It works just like i hoped. The only issue is, that for bigger files (200KB+) the hilighting is getting offset to the text it's marking :( 
Is there any way to fix it? Otherwise it's something we can live with :)

Thank you a lot for your help 
Cheers!
